packages/adp-tooling/src/base/node-generator-resolver.ts

```diff
@@ -0,0 +1,28 @@
+import { resolve } from 'path';
+
+/**
+ * Attempts to resolve the path to a specific node module generator from the NODE_PATH environment variable.
+ *
+ * @returns {string | undefined} The resolved path to the generator module if found, or undefined if not found.
+ */
+export function resolveNodeModuleGenerator(): string | undefined {
```

Comment:
What is this function doing and where do we use it?

Reply:
This is used in the prompt for extension project when the application is not supported by adaptation project. This is used in BAS to resolve the generator. It was the only option that worked with this.composeWith() of the yeoman generator. We get the path back to the generator (if found) from resolveNodeModuleGenerator. If no path is found we show a message that Extensibility Generator is not installed in the dev space and the user cannot continue.

Reply:
Could you enhance the jsdoc comment please was something similar to what you just wrote here.

Reply:
Changed in: [6d395b6](https://github.com/SAP/open-ux-tools/pull/2169/commits/6d395b63c1e16105d873c3bce51cc8646da5c0f1)